### PR TITLE
drivers: npmx: shell: add charging current getter

### PR DESCRIPTION
**Pull request by wilu-nordic**

### CODE REVIEW

File: drivers/npmx/npmx_shell.c [hunk 1]
```diff
@@ -3604,6 +3652,7 @@ static int cmd_vbusin_status_connected_get(const struct shell *shell, size_t arg
 /* Creating subcommands (level 3 command) array for command "adc meas". */
 SHELL_STATIC_SUBCMD_SET_CREATE(sub_adc_meas,
 			       SHELL_CMD(vbat, NULL, "Get battery voltage", cmd_adc_meas_vbat_get),
+			       SHELL_CMD(ibat, NULL, "Get charging current", cmd_adc_meas_ibat_get),
```
Comment (magp-nordic) [inline drivers/npmx/npmx_shell.c:3655]: `ibat` first (alphabetical order)

File: drivers/npmx/npmx_shell.c [hunk 2]
```diff
@@ -416,42 +416,90 @@ static bool check_pin_configuration_correctness(const struct shell *shell, int8_
 	return true;
 }
 
-static int cmd_adc_meas_vbat_get(const struct shell *shell, size_t argc, char **argv)
+static int adc_meas_vbat_get(const struct shell *shell, npmx_adc_meas_t adc_meas,
```
Comment (magp-nordic) [inline drivers/npmx/npmx_shell.c:419]:
```suggestion
static int adc_meas_get(const struct shell *shell, npmx_adc_meas_t adc_meas,
```

File: drivers/npmx/npmx_shell.c [hunk 3]
```diff
@@ -416,42 +416,90 @@ static bool check_pin_configuration_correctness(const struct shell *shell, int8_
...
+	}
+
+	if (!enable) {
+		/* Enable auto measurement of the battery current after
+		 * the battery voltage measurement. */
+		err_code = npmx_adc_ibat_meas_enable_set(adc_instance, true);
+		if (!check_error_code(shell, err_code)) {
+			print_set_error(shell, "charging current enable");
+			return 0;
+		}
```
Comment (masz-nordic) [inline drivers/npmx/npmx_shell.c:489]: Should it be disabled after getting the measurement?

Comment (wilu-nordic) [reply drivers/npmx/npmx_shell.c:489]: It is not required by an IPS.

File: drivers/npmx/npmx_shell.c [hunk 4]
```diff
@@ -416,42 +416,90 @@ static bool check_pin_configuration_correctness(const struct shell *shell, int8_
...
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	npmx_adc_t *adc_instance = adc_instance_get(shell);
+	if (adc_instance == NULL) {
+		return 0;
+	}
+
+	bool enable;
```
Comment (magp-nordic) [inline drivers/npmx/npmx_shell.c:475]:
```suggestion
	bool enabled;
```

**Review by magp-nordic — APPROVED** [review]
LGTM, I just left one note with a minor improvement

File: drivers/npmx/npmx_shell.c (hunk 4 above)
Comment (wilu-nordic) [reply drivers/npmx/npmx_shell.c]: Changed.

**Review by magp-nordic — APPROVED** [review]

**Comment by wilu-nordic** [comment]
Just rebased.

**Review by magp-nordic — APPROVED** [review]

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**